packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -538,3 +538,41 @@ export class OdspDocumentService implements IDocumentService {
         this.opsCache?.addOps(ops);
     }
 }
+
+export class LocalOdspDocumentService implements IDocumentService {
+    policies?: IDocumentServicePolicies | undefined;
+
+    constructor(
+        private readonly odspResolvedUrl: IOdspResolvedUrl,
+        private readonly logger: ITelemetryLogger,
+        private readonly fluidFile: Uint8Array | string,
+    ) {
+        this.policies = { storageOnly: true };
+    }
+
+    public get resolvedUrl(): IResolvedUrl {
+        return this.odspResolvedUrl;
+    }
+
+    public async connectToStorage(): Promise<IDocumentStorageService> {
+        return new LocalOdspDocumentStorageService(
+            this.odspResolvedUrl,
+            this.logger,
+            this.fluidFile,
+        );
+    }
+
+    public async connectToDeltaStorage(): Promise<IDocumentDeltaStorageService> {
+        // TODO: do we expect to call this?
```

Comment:
Yes, this call is expected. Snapshots contain trailing ops, and runtime gets to them through IDocumentDeltaStorageService.get().